### PR TITLE
switched AzureStaticWebApp to new swa client api and added custom build variable functionality

### PR DESCRIPTION
**Pull request by brhopcra**
**Task name**: AzureStaticWebAppV0

**Description**: Switched to new swa client api and added custom build variable functionality

**Documentation changes required:** N

**Added unit tests:** N

**Attached related issue:** Y https://github.com/Azure/static-web-apps/issues/392

**Checklist**:
- [x] Task version was bumped
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Review by miwebst — APPROVED** [review]

**Review by mishapos — APPROVED** [review]

File: Tasks/AzureStaticWebAppV0/index.ts [hunk 1]
```diff
@@ -54,4 +34,66 @@ async function run() {
...
+    const systemVariableNames: Set<string> = new Set<string>();
+
+    const addVariableToString = (variableName: string, variableValue: string) => variableString += variableName + "=" + variableValue + "\n"
+
+    const addSystemVariableToString = (variableName: string, variableValue: string) => {
+        addVariableToString(variableName, variableValue)
+        systemVariableNames.add(variableName)
+    }
+
+    const workingDirectory: string = tl.getInput('cwd', false) || "";
```
Comment (anthonychu) [inline Tasks/AzureStaticWebAppV0/index.ts:50]: Not sure if we need to support relative directories here as well. I think most folks are used to passing absolute paths so I think we're okay. Perhaps add a bit more to the help text to say that this is an absolute path.

File: Tasks/AzureStaticWebAppV0/index.ts [hunk 2]
```diff
@@ -19,39 +32,127 @@ async function run() {
...
         const options = {
             failOnStdErr: false
         };
 
         await bash.exec(<any>options);
         tl.setResult(tl.TaskResult.Succeeded, null);
-    }
-    catch (err) {
+    } catch (err) {
         tl.setResult(tl.TaskResult.Failed, null);
```
Comment (anthonychu) [inline Tasks/AzureStaticWebAppV0/index.ts:44]:
```suggestion
        tl.setResult(tl.TaskResult.Failed, err);
```

Print error when there's a failure

File: Tasks/AzureStaticWebAppV0/index.ts [hunk 3]
```diff
@@ -19,39 +32,127 @@ async function run() {
...
+    } finally {
+        await fs.promises.unlink(envVarFilePath);
```
Comment (anthonychu) [inline Tasks/AzureStaticWebAppV0/index.ts:46]:
```suggestion
        await fs.promises.unlink(envVarFilePath).catch(() => tl.debug("Unable to delete env file"));
```

If unable to delete the file, probably just need to log it (currently it throws an uncaught exception and fails the task).

**Review by anthonychu — COMMENTED** [review]
Looks like this is working great!

One thing that I found was that `envVarDenylist.json` needs to be added [here](https://github.com/microsoft/azure-pipelines-tasks/blob/master/make-options.json#L205) or the make script won't copy the file.

**Review by v-abajaj — APPROVED** [review]

**Review by v-abajaj — APPROVED** [review]

**Comment by miwebst** [comment]
@v-abajaj thank you for approving, can you help merge this PR?

**Comment by v-abajaj** [comment]
> @v-abajaj thank you for approving, can you help merge this PR?

Can you please ping me on Slack (v-bajaj) or teams?

**Review by miwebst — APPROVED** [review]

**Review by anthonychu — APPROVED** [review]

**Review by NAGA-BANDI — APPROVED** [review]

**Comment by bishal-pdMSFT** [comment]
Approving the PR based on @NAGA-BANDI and @anthonychu's approvals so that the PR merge can be unblocked.

**Review by bishal-pdMSFT — APPROVED** [review]

**Pull request closed without merging**